crates/client/rpc/src/lib.rs

```diff
@@ -1563,6 +1562,42 @@ where
         Ok(MaybePendingTransactionReceipt::Receipt(receipt))
     }
 
+    fn get_transaction_execution_info(
+        &self,
+        parent_substrate_block_hash: B::Hash,
+        block_transactions: &BlockTransactions,
+        transaction_hash: TransactionHash,
+    ) -> Result<TransactionExecutionInfo, StarknetRpcApiError>
+    where
+        B: BlockT,
+    {
+        let (transactions_before, transaction_to_trace) =
+            split_block_tx_for_reexecution(block_transactions, transaction_hash).map_err(|e| {
+                log::error!("Failed to split block transactions for re-execution: {e}");
+                StarknetRpcApiError::InternalServerError
+            })?;
+
+        if transaction_to_trace.is_empty() {
+            return Err(StarknetRpcApiError::TxnHashNotFound);
+        }
+
+        if transaction_to_trace.len() > 1 {
+            log::error!("More than one transaction with the same transaction hash {:#?}", transaction_to_trace);
+            return Err(StarknetRpcApiError::InternalServerError);
+        }
+
+        let mut trace = self
+            .re_execute_transactions(parent_substrate_block_hash, transactions_before, transaction_to_trace)
+            .map_err(|e| {
+                log::error!("Failed to re-execute transactions: {e}");
+                StarknetRpcApiError::InternalServerError
+            })?;
+
+        let execution_info = trace.remove(0);
+
+        Ok(execution_info.0)
```

Comment:
`execution_info.1` are the state diffs, computed here:
crates/pallets/starknet/src/simulations.rs
```rust
let res = res.map(|r| (r, transactional_state.to_state_diff()));
``` 

We don't need them here, so it's wasted execution. Can you add a bool arg to `re_execute_transaction` runtime-api method to decide if we want the stateDiff, and change the signature so that the second ret value is an `Option`

Reply:
fixed